chain-signatures/node/src/protocol/message/mod.rs

```diff
@@ -819,18 +811,19 @@ pub struct MessageOutbox {
 }
 
 impl MessageOutbox {
-    pub fn new(client: NodeClient, id: &AccountId) -> Self {
+    pub fn new(id: &AccountId, client: NodeClient, outbox_rx: mpsc::Receiver<SendMessage>) -> Self {
         Self {
             client,
-            messages: HashMap::new(),
             account_id: id.clone(),
+            outbox_rx,
+            messages: HashMap::new(),
         }
     }
 
-    pub fn extend(&mut self, outgoing: &mut mpsc::Receiver<SendMessage>) {
+    pub fn extend(&mut self) {
```

Comment:
Again, an `extend` method without an argument looks very un-rusty to my eye :shrug: